lib/trace-agent.js

```diff
@@ -162,9 +162,15 @@ TraceAgent.prototype.addTransactionLabel = function(key, value) {
 /**
  * Determines whether a trace of the given name should be recorded based
  * on the current tracing policy.
+ *
+ * @param {string} name the url to trace
+ * @param {number} options the trace header options
```

Comment:
Actually, here's the path on which it is undefined: when parseContextFromHeader returns null, we assign `traceHeader` to `{}`.